addons/sale/models/sale_order_line.py

```diff
@@ -964,10 +964,13 @@ def _add_precomputed_values(self, vals_list):
 
         This is a temporary fix until the problem is fixed in the ORM.
         """
-        precision = self.env['decimal.precision'].precision_get('Discount')
+        discount_precision = self.env['decimal.precision'].precision_get('Discount')
+        quantity_precision = self.env['decimal.precision'].precision_get('Product Unit of Measure')
         for vals in vals_list:
             if vals.get('discount'):
```

Comment:
```suggestion
            for fname in ('discount', 'product_uom_qty'):
                 if fname in vals:
                     vals[fname] = self._fields[fname].convert_to_cache(vals[fname], self)
```
Could be a good opportunity to delegate the precision & rounding management to the ORM as much as possible ?

Reply:
nice, I've updated the PR, thanks :+1: